dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -12,19 +12,46 @@
 
   {{ run_hooks(pre_hooks) }}
 
-  -- cleanup
-  {%- if old_relation is not none -%}
-    {{ drop_relation(old_relation) }}
-  {%- endif -%}
+  {%- if old_relation is none or table_type != 'iceberg' -%}
+    {%- if old_relation is not none -%}
+      {{ drop_relation(old_relation) }}
+    {%- endif -%}
 
-  -- build model
-  {% call statement('main') -%}
-    {{ create_table_as(False, target_relation, sql) }}
-  {%- endcall %}
+    {%- call statement('main') -%}
+      {{ create_table_as(False, target_relation, sql) }}
+    {%- endcall %}
 
-  {% if table_type != 'iceberg' %}
-    {{ set_table_classification(target_relation) }}
-  {% endif %}
+    {%- if table_type != 'iceberg' -%}
+      {{ set_table_classification(target_relation) }}
+    {%- endif -%}
+  {%- else -%}
+    {%- set tmp_relation = api.Relation.create(identifier=target_relation.identifier ~ '__ha',
```

Comment:
how about using `__dbt_tmp` instead of `__ha` to be consitent with other tmp table conventions used in the adapter.

Reply:
We use `__ha` for `table_hive_ha` so I went for the same naming convention.